MH4U Dex/CoreDataController.m

```diff
@@ -363,6 +439,87 @@ - (NSUInteger)countForEntityWithEntityName:(NSString *)entityName withPredicate:
     }
 }
 
+- (QuestDrop *)questDropFromDictionary:(NSDictionary *)dropDict
+{
+    QuestDrop *drop = [QuestDrop insertInManagedObjectContext:self.managedObjectContext];
+    NSPredicate *questPredicate = [NSPredicate predicateWithFormat:@"%K == %@", QuestAttributes.id, dropDict[QuestDropQuestIDKey]];
+    drop.quest = (Quest *)[self getUniqueEntityWithEntityName:[Quest entityName] withPredicate:questPredicate];
+    if (!drop.quest) {
+        //If the quest was not found, then it makes no sense to store this reward.  Just return nil.
+        NSLog(@"The quest for a quest reward was not found!");
+        return nil;
+    }
+    drop.id = dropDict[QuestDropIDKey];
+    drop.row = dropDict[QuestDropRowKey];
+    drop.quantity = dropDict[QuestDropQuantityKey];
+    drop.percent = dropDict[QuestDropPercentKey];
+    NSPredicate *itemPredicate = [NSPredicate predicateWithFormat:@"%K == %@", ItemAttributes.name, dropDict[QuestDropItemNameKey]];
+    Item *item = (Item *)[self getUniqueEntityWithEntityName:[Item entityName] withPredicate:itemPredicate];
+    if (!item) {
+        item = [Item insertInManagedObjectContext:self.managedObjectContext];
+        item.name = dropDict[QuestDropItemNameKey];
+    }
+    drop.item = item;
+    return drop;
+}
+
+- (Quest *)questFromDictionary:(NSDictionary *)questDict
+{
+    Quest *quest = [Quest insertInManagedObjectContext:self.managedObjectContext];
+    quest.id = questDict[QuestIDKey];
+    quest.name = questDict[QuestNameKey];
+    if ([questDict[QuestCaravanHallKey] isEqualToString:QuestCaravanKey]) {
```

Comment:
Why are these coming through as `NSString` objects in the JSON rather than `NSNumber` objects with boolean values?